### ISSUE TITLE
Unable to detect storage account network rules differences between configuration and actual Azure resource

### ISSUE THREAD

**Issue by liemnotliam** [terraform-providers/terraform-provider-azurerm#1412]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version
Terraform v0.11.7
+ provider.azurerm v1.7.0

### Affected Resource(s)

* azurerm_storage_account

### Terraform Configuration Files

The bug happens when either `ip_rules` or `virtual_network_subnet_ids` are empty. `ip_rules` are set as empty for this example.

```hcl
resource "azurerm_resource_group" "test" {
  name     = "test-rg"
  location = "eastus"
}

resource "azurerm_virtual_network" "test" {
  name                = "test-vnet"
  address_space       = ["10.0.0.0/16"]
  location            = "${azurerm_resource_group.test.location}"
  resource_group_name = "${azurerm_resource_group.test.name}"
}

resource "azurerm_subnet" "test" {
  name                 = "subnet"
  resource_group_name  = "${azurerm_resource_group.test.name}"
  virtual_network_name = "${azurerm_virtual_network.test.name}"
  address_prefix       = "10.0.2.0/24"
  service_endpoints    = ["Microsoft.Storage"]
}

resource "azurerm_storage_account" "test" {
  name                = "luewabsw12"
  resource_group_name = "${azurerm_resource_group.test.name}"

  location                 = "${azurerm_resource_group.test.location}"
  account_tier             = "Standard"
  account_replication_type = "LRS"

  network_rules {
    ip_rules                   = []
    virtual_network_subnet_ids = ["${azurerm_subnet.test.id}"]
  }
}

```

### Expected Behavior
Given the same Terraform configuration, if there are changes in storage account network rules, then Terraform should detect them. Given the example configuration above, if the IP rules remain empty and the virtual network rules differ from the Terraform configuration then Terraform should detect the change. Currently it doesn't due to the checks [here](https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/resource_arm_storage_account.go#L674).

### Actual Behavior
No changes found between resource configuration and the actual Azure storage account.
```
No changes. Infrastructure is up-to-date.

This means that Terraform did not detect any differences between your
configuration and real physical resources that exist. As a result, no
actions need to be performed.
```

### Steps to Reproduce

1. `terraform apply`
2. Update network rules subnets to be different from terraform configuration via azure CLI or portal (e.g. remove subnet from network rules)
3. `terraform plan`

If an IP rule is added, then Terraform detects the changes as it satisfies the [if-checks when reading the storage account properties](https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/resource_arm_storage_account.go#L674).

**Comment by mbfrahry** [comment]
@liemnotliam, this issue should be resolved with #1613 and it will be going out in the next release. Feel free to reopen this issue or open another issue with any further questions.

**Comment by tombuildsstuff** [comment]
hi @luisamador 

Just to let you know that is now available in [v1.10 of the AzureRM Provider(https://github.com/terraform-providers/terraform-provider-azurerm/blob/v1.10.0/CHANGELOG.md#1100-july-21-2018) - which you can update to this version by specifying it in the Provider block, like so:

```
provider "azurerm" {
  version = "=1.10.0"
}
```

Thanks!
